core/dbt/graph/selector_spec.py

```diff
@@ -173,12 +174,14 @@ class BaseSelectionGroup(dbtClassMixin, Iterable[SelectionSpec], metaclass=ABCMe
     def __init__(
         self,
         components: Iterable[SelectionSpec],
+        indirect_selection: IndirectSelection = IndirectSelection.Eager,
```

Comment:
In case this is confusing - in `selector.py` on line 138 we need to get the indirect selection method from a composite spec (a union, difference, or intersection). This change here, and the changes in cli.py, add the indirect selection method to these composite specs through the BaseSelectionGroup  so we can get them at that line in selector.py. Previously the indirect selection method was only on the SelectionCriteria object.

It's possible I misunderstood how something works here, but I couldn't find an easier way to get the indirect selection method at that point.

Reply:
Thanks for this note, really helpful!